### PR TITLE
This allows bitcode files as inputs when using -c and the llvm wasm b…

### PR DESCRIPTION
**Pull request by vargaz**
This allows bitcode files as inputs when using -c and the llvm wasm backend, i.e.
```emcc -o foo.o -c foo.bc```
will compile the bc file into a wasm object file.
This is useful for compiling .bc files created by 3rd party compilers using the same flags that emcc
passes to clang, so they can be linked together later.

### CODE REVIEW

**Comment by vargaz** [comment]
This is a new version of:
https://github.com/emscripten-core/emscripten/pull/8457
It includes a test.

File: emcc.py [hunk 1]
```diff
@@ -1010,7 +1010,8 @@ def add_link_flag(i, f):
         file_suffix = get_file_suffix(arg)
         if file_suffix in SOURCE_ENDINGS + OBJECT_FILE_ENDINGS + DYNAMICLIB_ENDINGS + ASSEMBLY_ENDINGS + HEADER_ENDINGS or shared.Building.is_ar(arg): # we already removed -o <target>, so all these should be inputs
           newargs[i] = ''
-          if file_suffix.endswith(SOURCE_ENDINGS):
+          #if file_suffix.endswith(SOURCE_ENDINGS):
```
Comment (kripken) [inline emcc.py]: these unnecessary comments need to be removed for `flake8` to pass.

File: emcc.py [hunk 2]
```diff
@@ -1010,7 +1010,8 @@ def add_link_flag(i, f):
...
+          if file_suffix.endswith(SOURCE_ENDINGS) or (has_dash_c and file_suffix.endswith(OBJECT_FILE_ENDINGS)):
```
Comment (sbc100) [inline emcc.py]: Hmm.. my outstanding change is about to get rid of OBJECT_FILE_ENDINGS.   Should this perhaps be "is_bitcode"? or check bitcode endings specifically?   We should do whatever clang does I guess.  Does it allow `.o` as an input to compiler, or just `.bc`?
 
See https://github.com/emscripten-core/emscripten/pull/10560/

File: emcc.py (hunk 1 above)
Comment (vargaz) [reply emcc.py]: Should be fixed.

File: emcc.py (hunk 2 above)
Comment (vargaz) [reply emcc.py]: It seems to only accept .bc, for .o, it says:
clang-11: warning: hello.exe.o: 'linker' input unused [-Wunused-command-line-argument]

Comment (sbc100) [reply emcc.py]: In that case maybe just use ".bc" rather than OBJECT_FILE_ENDINGS?

Comment (vargaz) [reply emcc.py]: Done.

File: emcc.py [hunk 3]
```diff
@@ -1010,7 +1010,7 @@ def add_link_flag(i, f):
         file_suffix = get_file_suffix(arg)
         if file_suffix in SOURCE_ENDINGS + OBJECT_FILE_ENDINGS + DYNAMICLIB_ENDINGS + ASSEMBLY_ENDINGS + HEADER_ENDINGS or shared.Building.is_ar(arg): # we already removed -o <target>, so all these should be inputs
           newargs[i] = ''
-          if file_suffix.endswith(SOURCE_ENDINGS):
+          if file_suffix.endswith(SOURCE_ENDINGS) or (has_dash_c and file_suffix.endswith('.bc')):
```
Comment (sbc100) [inline emcc.py]: I'm not sure why `endswith` is used in the places near here but you can just do `file_suffix == '.bc'`.

**Pull request merged**